internal/db/schema/migrations/postgres/2/90_auth.up.sql

```diff
@@ -0,0 +1,97 @@
+begin;
+
+-- By adding the name column to the base auth method type, the database can
+-- ensure that auth method names are unique across all sub types.
+alter table auth_method
+  add column name wt_name;
+
+alter table auth_method
+  add constraint auth_method_scope_id_name_uq
+    unique (scope_id, name);
+
+
+-- the intent of this statement is to update the base type's name with the
+-- existing password auth method names.
+update auth_method 
+set name = pw.name
+from 
+  auth_password_method pw
+where 
+  auth_method.public_id = pw.public_id and
+  pw.name is not null;
+
+-- insert_auth_method_subtype() is a replacement of the function definition in
+-- migration 07_auth.up.sql  This new definition also inserts the sub type's name
+-- into the base type. The name column must be on the base type, so the database
+-- can ensure that auth method names are unique across all sub types.
+create or replace function
+  insert_auth_method_subtype()
```

Comment:
Should there be `create trigger` calls for this function for `auth_password_method` and `auth_oidc_method`?

Reply:
there were already triggers for this function on those tables.  this just replaces the function's definition.